docs/design/security/README.adoc

```diff
@@ -0,0 +1,534 @@
+in the realm configuration.
+
+[source,yaml]
+----
+hazelcast:
+  security:
+    enabled: true
+    realms:
+      - name: simpleRealm-clients
+        authentication:
+          simple:
+            users:
+              - username: root
+                password: 7931verySecret
+                roles:
+                  - admin
+              - username: readonly
+                password: anotherSecret654
+                roles:
+                  - monitor
+    # map the client protocol authentication to the defined realms
+    client-authentication:
+      realm: simpleRealm-clients
+    client-permissions:
+      all:
+        principal: admin
+----
+
+[[client-access-control-basics]]
+== Client access control
+
+When the security is enabled, permission checks are done on client
+requests. You have to grant permissions to named roles to allow clients
+to execute actions on a member.
+
+NOTE: Operations executed directly on members (in embedded mode) or the ones executed
+within member-to-member communication are without permission checks.
+
+[source,yaml]
+----
+hazelcast:
+  security:
+    enabled: true
+    client-permissions:
+      # local clients with "admin" role assigned have full access to all operations
+      all:
+        principal: admin
+        endpoints:
+          - 127.0.0.1
+      map:
+        # every authenticated client can work with a map named "playground"
+        - name: playground
+          actions:
+            - all
+        # clients with role "monitor" can read from a map named "accounts"
+        - name: accounts
+          principal: monitor
+          actions:
+            - read
+        # clients with role "application" can work with the "accounts" 
+        # map if they come from given network segments
+        - name: accounts
+          principal: application
+          endpoints:
+            - 192.168.1.*
+            - 192.168.2.*
+          actions:
+            - create
+            - destroy
+            - put
+            - read
+            - remove
+            - lock
+----
+
+=== Client permissions
+
+Permission has:
+
+* type (e.g. `map permission`)
+* name - usually the data structure name (e.g. `employees`)
+* actions - operations allowed (e.g. `read`, `write`)
+* endpoints - remote client IP addresses to which the permission is
+granted (e.g. `192.168.\*.*`)
+
+Wildcards (`*`) are supported in names and endpoint definitions.
+
+== Security realms
+
+The security realm is a named security configuration. It's intended mainly
+for member configuration but with limited support on Java clients too.
+Security realms in current Hazelcast versions allow defining identity
+and authentication configurations.
+
+== Advanced security topics
+
+=== Auditlog
+
+Hazelcast Auditlog is a security feature that records and logs
+security-related events and operations within the Hazelcast cluster,
+providing increased visibility and traceability for monitoring and
+compliance purposes.
```

Comment:
Is there a default auditlog impl? If so can we add an example of that here? If not, we already mention about the SPI below so maybe we can remove this part?

Reply:
:+1: I'll add a short description of the default implementation.